src/mongocxx/test_util/client_helpers.cpp

```diff
@@ -248,6 +248,11 @@ bool is_replica_set(const client& client) {
     return static_cast<bool>(reply.view()["setName"]);
 }
 
+bool is_sharded_cluster(const client& client) {
+    auto reply = get_is_master(client);
+    return static_cast<bool>(reply.view()["config"]["sharded"]);
```

Comment:
Suggest using the check from [Server Discovery and Monitoring](https://github.com/mongodb/specifications/blob/ba44473e2488290e13acabbea0f32fa8c2bc4b7a/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#type).

```suggestion
    if (!reply.view()["msg"]) {
        return false;
    }
    return 0 == reply.view()["msg"].get_string().value.compare ("isdbgrid");
```

This is also used in the C driver [test_framework_is_mongos](https://github.com/mongodb/mongo-c-driver/blob/0636093de1e11b1259f75f21706b906bb11ae382/src/libmongoc/tests/test-libmongoc.c#L2030-L2046) function.